src/watch.rs

```diff
@@ -95,36 +107,43 @@ where
 
     let (tx_changes, rx_changes) = channel();
 
-    start_event_thread(rx_sources, tx_changes.clone(), ChangeType::Sources);
-    start_event_thread(rx_code, tx_changes, ChangeType::Code);
+    start_event_thread(
+        rx_sources,
+        tx_changes.clone(),
+        ChangeType::Sources,
+        suspend.clone(),
+    );
+    start_event_thread(rx_code, tx_changes, ChangeType::Code, suspend);
 
     Ok((rx_changes, source_watcher, code_watcher))
 }
 
 fn start_event_thread(
-    in_channel: Receiver<DebouncedEvent>,
+    in_channel: Receiver<RawEvent>,
     out_channel: Sender<ChangeType>,
     event_type: ChangeType,
+    suspend: Arc<AtomicBool>,
 ) {
     std::thread::spawn(move || loop {
         let mut send_event = false;
 
-        let event = in_channel.recv().unwrap();
-        if is_file_change(event) {
+        in_channel.recv().unwrap();
+        if !suspend.load(Ordering::SeqCst) {
             send_event = true;
         }
 
-        let deadline = Instant::now() + Duration::from_millis(COLLECT_EVENTS_MILLIS);
+        let mut deadline = Instant::now() + Duration::from_millis(COLLECT_EVENTS_MILLIS);
         loop {
             let timeout = match deadline.checked_duration_since(Instant::now()) {
```

Comment:
I think you can drop all this, you just need
```rust
while in_channel.recv_timeout(COLLECT_EVENTS).is_ok() {}
```
now. (It should be possible to have a `const COLLECT_EVENTS: Duration = Duration::from_millis(500)` as `from_millis` is a `const fn` by now.)

Reply:
Since you are not filtering either anymore, I guess this can be simplified further still, e.g.
```
loop {
  in_channel.recv().unwrap();
  if suspend.load(Ordering::SeqCst) { continue; }

  while in_channel.recv_timeout(COLLECT_EVENTS).is_ok() {}
  if suspend.load(Ordering::SeqCst) { continue; }

  out_channel.send(event_type).unwrap();
}
```

Reply:
🎉